### PR TITLE
Migrate from circleci to Jenkins

### PR DESCRIPTION
**Pull request by shawkinsmezmo**
Implements a Jenkins pipeline for the logspout repo. The jenkins pipeline performs the following actions:
* Lint the code
* build the image
* Invoke semver management
* Push docker image to dockerhub

The semver is managed by [svu](https://github.com/caarlos0/svu)
The PR also contains:
* Removes existing circleci config
* Reformats the code according to the output of `gofmt`. This is necessary because the code previously was not formatted correctly and would not pass gofmt.

The pipeline has been tested as follows:
* Tested that non-master commits invoke a Jenkins build.
* Tested that non-master commits do not invoke undesired functionality (creating release, pushing images) and only perform linting step
* Tested that `Create Release` step correctly creates and manages the semantic version
* Tested that docker images are built successfully and tagged appropriately according to the release

### CODE REVIEW

**Review by gjanco — DISMISSED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      }
+
+      stages {
+        stage('Create Release') {
+          tools {
+            nodejs 'NodeJS 14'
+          }
+
+          steps {
+            versioner(
```
Comment (charlesthomas) [inline Jenkinsfile:64]: curious why you ended up using semantic-release

based on our slack conversation, i thought you were going to use `svu`

File: Makefile [hunk 2]
```diff
@@ -1,13 +1,48 @@
-TAG          = $(if $(CIRCLE_TAG),$(CIRCLE_TAG),PRTEST)
+BUILD_ARCH  := $(shell uname -m)
+BUILD_OS    := $(shell uname -s | tr '[:upper:]' '[:lower:]')
+TAG          = $(shell VER=$$(git tag | tail -1); echo $${VER:1})
```
Comment (charlesthomas) [inline Makefile:3]: i don't think this is going to work in jenkins because of the way it "clones"

Comment (shawkinsmezmo) [reply Makefile:3]: There is a `git fetch --tags` that is executed prior to the `TAG` variable being evaluated, so this works as expected. Here's an example from my testing - https://jenkins.use.int.logdna.net/blue/organizations/jenkins/logdna%2Flogspout/detail/PR-41/67/pipeline/

**Review by charlesthomas — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (dkhokhlov) [reply Makefile:3]: for  PR builds this TAG will have default val?
release dry run might be useful as part of CI test.

Comment (shawkinsmezmo) [reply Makefile:3]: The only stage that executes currently for PR builds is the lint stage, so this shouldn't ever be a problem.

**Pull request merged**